### PR TITLE
Screenshot improvements

### PR DESCRIPTION
**Pull request by Hixie**
Add an "s" command to `flutter run` which takes a screenshot.
Make that comment turn off the slow mode banner.

### CODE REVIEW

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/5021

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/7772

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 1]
```diff
@@ -109,6 +106,24 @@ abstract class ResidentRunner {
     await currentView.uiIsolate.flutterToggleDebugPaintSizeEnabled();
   }
 
+  Future<Null> _screenshot() async {
+    File outputFile = getUniqueFile(fs.currentDirectory, 'flutter', 'png');
+    try {
+      if (vmService != null)
+        await vmService.vm.refreshViews();
+      if (isRunningDebug)
+        await currentView.uiIsolate.flutterDebugAllowBanner(false);
```
Comment (xster) [inline packages/flutter_tools/lib/src/resident_runner.dart]: Should we try to extract the logic here and in screenshot.dart to share nice features like these?

**Review by xster — APPROVED** [review]
Super awesome!
LGTM

**Comment by sethladd** [comment]
Very cool :)

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter_tools/lib/src/resident_runner.dart]: The problem is that screenshot.dart has a very different context. It doesn't have a VMService connection for example. We could provide a way to specify the observatory port, etc, but that seems more work than this is worth.

**Pull request merged**